rootfs/scheduler/__init__.py

```diff
@@ -259,13 +261,14 @@ def _get_deploy_batches(self, steps, desired):
 
         return batches
 
-    def _update_application_service(self, namespace, name, app_type, port, routable=False):
+    def _update_application_service(self, namespace, name, app_type, port, routable=False, maintenance=False):  # noqa
         """Update application service with all the various required information"""
         service = self.get_service(namespace, namespace).json()
         old_service = service.copy()  # in case anything fails for rollback
 
         try:
             # Update service information
+            service['metadata']['annotations']['router.deis.io/maintenance'] = str(maintenance)
```

Comment:
if the user sets the appsettings first and then does a deploy set_maintenance doesn't get called..set_maintenance is called only during the save of the record and if the value is changed.